packages/core/src/layout/SignInPage/providers.tsx

```diff
@@ -15,37 +15,60 @@
  */
 
 import React, { useLayoutEffect, useState, useMemo, useCallback } from 'react';
-import { guestProvider } from './guestProvider';
-import { googleProvider } from './googleProvider';
-import { customProvider } from './customProvider';
-import { gitlabProvider } from './gitlabProvider';
-import { oktaProvider } from './oktaProvider';
-import { githubProvider } from './githubProvider';
 import {
   SignInPageProps,
   SignInResult,
   useApi,
   useApiHolder,
   errorApiRef,
 } from '@backstage/core-api';
-import { SignInProvider } from './types';
+import { SignInConfig, IdentityProviders, SignInProvider } from './types';
+import { commonProvider } from './commonProvider';
+import { guestProvider } from './guestProvider';
+import { customProvider } from './customProvider';
 
 const PROVIDER_STORAGE_KEY = '@backstage/core:SignInPage:provider';
 
-// Separate list here to avoid exporting internal types
-export type SignInProviderId = 'guest' | string;
+export type SignInProviderId = 'guest' | 'custom' | string;
```

Comment:
No need for a separate type here imo, can just use `string` in the locations this was being used.